### PR TITLE
bt-707: Add placeholder for empty candidates search result

### LINKED ISSUE #707: Add placeholder "No candidates were found, try to change the filtering" to the Search field when no matches
Image

### PR DESCRIPTION
**Pull request by vladjsp**
closes https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/issues/707

https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/118971905/14b0eb99-6648-4629-80c5-f7bc8c4ff37b

### CODE REVIEW

File: frontend/src/bundles/search-candidates/pages/candidates.tsx [hunk 1]
```diff
@@ -162,16 +162,30 @@ const Candidates: React.FC = () => {
...
+                        {filteredCandidates.length > 0 ? (
+                            filteredCandidates.map((candidate) => (
+                                <CandidateProfile
+                                    key={candidate.id}
+                                    isProfileCard
+                                    candidateData={candidate}
+                                />
+                            ))
+                        ) : (
+                            <Grid container item className={styles.noResults}>
```
Comment (nikita-remeslov) [inline frontend/src/bundles/search-candidates/pages/candidates.tsx]: looks like redundunt

File: frontend/src/bundles/search-candidates/pages/candidates.tsx [hunk 2]
```diff
@@ -162,16 +162,30 @@ const Candidates: React.FC = () => {
...
+                                <Typography
+                                    className={styles.noResultsText}
+                                    variant="body1"
+                                >
+                                    {' '}
+                                    No candidates were found, try to change the
+                                    filtering{' '}
```
Comment (nikita-remeslov) [inline frontend/src/bundles/search-candidates/pages/candidates.tsx]: why do we need `{' '}`?

File: frontend/src/bundles/search-candidates/pages/candidates.tsx [hunk 3]
```diff
@@ -162,16 +162,30 @@ const Candidates: React.FC = () => {
...
                         )}
                     >
-                        {filteredCandidates.map((candidate) => (
-                            <CandidateProfile
-                                key={candidate.id}
-                                isProfileCard
-                                candidateData={candidate}
-                            />
-                        ))}
+                        {filteredCandidates.length > 0 ? (
```
Comment (nikita-remeslov) [inline frontend/src/bundles/search-candidates/pages/candidates.tsx:169]:
```suggestion
                        {!filteredCandidates.length ? (
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/search-candidates/pages/candidates.tsx (hunk 3 above)
Comment (vladjsp) [reply frontend/src/bundles/search-candidates/pages/candidates.tsx:169]: these changes will cause eslint unicorn/no-negated-condition and unicorn/explicit-length-check errors.

File: frontend/src/bundles/search-candidates/pages/candidates.tsx (hunk 1 above)
Comment (vladjsp) [reply frontend/src/bundles/search-candidates/pages/candidates.tsx]: fixed

File: frontend/src/bundles/search-candidates/pages/candidates.tsx (hunk 2 above)
Comment (vladjsp) [reply frontend/src/bundles/search-candidates/pages/candidates.tsx]: fixed. 
strange, I didn't write that. maybe vs code inserted it for some reason

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**